cli/command/image/build.go

```diff
@@ -192,7 +201,7 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 		var excludes []string
 		if err == nil {
 			excludes, err = dockerignore.ReadAll(f)
-			if err != nil {
+			if !ignoreErr && err != nil {
```

Comment:
Is this the right spot for this? if we get an os.IsNotExist() above then won't `f` be nil and then the `ReadAll()` will return immediately with `nil, nil`?  It seems as though `ignoreErr` might be needed at all, or you'll want to add it to the `if err == nil {` line (2 lines above) to skip the ReadAll() in that case.  But, it actually seems like that "if" might be better written as `if f != nil {`